addons/mail/models/mail_thread.py

```diff
@@ -1551,9 +1551,12 @@ def message_post(self, body='', subject=None, message_type='notification',
         # 4: mail.message.subtype
         subtype_id = False
         if subtype:
-            if '.' not in subtype:
-                subtype = 'mail.%s' % subtype
-            subtype_id = self.env['ir.model.data'].xmlid_to_res_id(subtype)
+            if isinstance(subtype, (str, unicode)):
```

Comment:
Note to myself: revert that code and directly gives subtype_id to message_post .